blockchain/blockchain.go

```diff
@@ -634,10 +632,18 @@ func (b *Blockchain) WriteHeadersWithBodies(headers []*types.Header) error {
 		}
 	}
 
-	// Write the actual headers
+	// Write the actual headers in separate batches for now
 	for _, h := range headers {
 		event := &Event{}
-		if err := b.writeHeaderImpl(event, h); err != nil {
+
+		batch := b.db.NewBatch()
+		batchHelper := storage.NewBatchHelper(batch)
+
+		if err := b.writeHeaderImpl(batchHelper, event, h); err != nil {
+			return err
+		}
+
+		if err := batch.Write(); err != nil {
```

Comment:
Seems like this can be organized a bit better.

I'd consider letting `batchHelper` to manipulate with all the batch operations. So in this case, `Write` could be called against `batchHelper` (consider maybe invoking it even inside `writeHeaderImpl`).

Reply:
@igorcrevar I made this change. Can you please take a look? 

Pull Request CI / Tests that are failing are passing locally. I'll run CI again after E2E tests are finished.

Reply:
`BatchHelper` is renamed to `BatchWriter` and all write operations are now going through him - `KeyValueStorage` is used for reading operations and to retrieve `Batch` implementation used by `BatchWriter`